Android/app/src/main/java/app/intra/socks/OverrideSocksHandler.java

```diff
@@ -70,10 +71,13 @@ public void doConnect(Session session, CommandMessage commandMessage) throws IOE
     // set default bind address.
     byte[] defaultAddress = {0, 0, 0, 0};
     bindAddress = InetAddress.getByAddress(defaultAddress);
+    int tcpHandshakeMs = -1;
     // DO connect
     try {
       // Connect directly.
+      long beforeConnect = SystemClock.elapsedRealtime();
       socket = new Socket(remoteServerAddress, remoteServerPort);
+      tcpHandshakeMs = (int)(SystemClock.elapsedRealtime() - beforeConnect);
```

Comment:
On socket errot, this will never be set. Is that ok?

Reply:
That seems OK to me.